### DISCUSSION TITLE
How the clan capital contributions are fetched

### DISCUSSION THREAD

**Discussion by thunder-07** [ClashKingInc/ClashKingBot#62] [Q&A]
I just joined a clan and seen this app on their discord server I'm surprised to find that this app is posting the clan gold contributions every time someone contributes are there any webhooks ? I mean does clash of clans api tells you when something like this happens are we periodically fetch the api to check?.

**Comment by Joelsuperstar** [comment]
yeah. You send a request to the clash of clans API for the player stats. 1 of them is capital contributions. You can save the last contribution amount. And you can notice a differents if someone had donated. Clash of clans has like a cache system. So you have to check like every 10 minutes for a change :)

**Reply by thunder-007** [reply]
@Joelsuperstar thanks for the reply bro could you show me the code where it's done

**Reply by Joelsuperstar** [reply]
There is documentation of the clash of clans API on the site of clash of clans
[https://developer.clashofclans.com/#/documentation](https://developer.clashofclans.com/#/documentation)

**Reply by MagicTheDev** [reply]
https://github.com/ClashKingInc/ClashKingTracking/blob/master/bot/player/track.py

**Reply by thunder-007** [reply]
@MagicTheDev  cool this keeps fetching the data periodically right instead is there an efficient way to do it like event driven web hooks for clan requests and capital gold contributions. I mean is there some option in clash of clans api to have callbacks.

**Reply by MagicTheDev** [reply]
There are no websockets, just polling the api

**Reply by thunder-007** [reply]
Thanks I'll try raising a support ticket to clash of clans and ask them include call backs and web hooks for clan requests and contributions but anyhow the product is awesome even with polling it's very reliable.

**Reply by Joelsuperstar** [reply]
@thunder-007 good luck in that. The API is really the lowest on the prio list for clash of clans, i guess. The API is still the same in ages. No really big updates. You can try but i give you a 0% change haha
